microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/PropertiesManager.java

```diff
@@ -109,36 +111,104 @@ public MicroProfileProjectInfo getMicroProfileProjectInfo(IJavaProject javaProje
 		if (LOGGER.isLoggable(Level.INFO)) {
 			LOGGER.info("Start computing MicroProfile properties for '" + info.getProjectURI() + "' project.");
 		}
+		SubMonitor mainMonitor = SubMonitor.convert(monitor,
+				"Collecting MicroProfile properties for '" + javaProject.getProject().getName() + "' project", 100);
 		try {
-			// Get the java project used for the search
 			boolean excludeTestCode = classpathKind == ClasspathKind.SRC;
-			IJavaProject javaProjectForSearch = getJavaProject(javaProject, excludeTestCode, monitor);
 
-			// Create JDT Java search pattern, engine and scope
+			// Step1 (50%) : get the java project used for the search
+			IJavaProject javaProjectForSearch = configureSearchClasspath(javaProject, excludeTestCode, mainMonitor);
+			if (mainMonitor.isCanceled()) {
+				throw new OperationCanceledException();
+			}
+
+			// Step2 (50%) : scan Java classes from the search classpath

```

Comment:
it'd be clearer to create the 50% submonitor here

Reply:
@fbricon updated